File: level2/구명보트.py
def solution(people, limit):
    people.sort()
    
    answer = 0
    s, e = 0, len(people)-1
    while s <= e:
        answer += 1
        if people[s]+people[e] <= limit:
            s += 1
        e -= 1
            
    return answer

File: level2/구명보트.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<int> people, int limit) {
    int left = 0, right = people.size()-1;
    sort(people.begin(),people.end());
    
    int answer = 0;
    while(left <= right){
        if(left == right){
            answer++;
            break;
        }
        if(people[left]+people[right] <= limit){
            answer++;
            left++;
            right--;
        }
        else{
            answer++;
            right--;
        }
    }
    return answer;
}
